plenum/server/replica.py

```diff
@@ -447,7 +447,11 @@ def on_view_change_done(self):
             self.last_prepared_before_view_change = None
         self.stasher.process_all_stashed(STASH_VIEW)
 
+    def _clear_all_3pc_msgs_after_vc(self):
+        self._ordering_service._clear_all_3pc_msgs_after_vc()
+
     def clear_requests_and_fix_last_ordered(self):
+        self._clear_all_3pc_msgs_after_vc()
```

Comment:
Why do we need this new code if already call `gc` in `_gc_before_new_view`?

Reply:
_gc_before_new_view will be called on primaryName changing. On starting view_change we set primaryName to None and _clear_all_3pc_msgs_after_vc does not needed before view_change